go/dpagg/mean.go

```diff
@@ -213,6 +215,101 @@ func (bm *BoundedMeanFloat64) Result() float64 {
+			return noise.ConfidenceInterval{}, err
+		}
+		tightness := confInt.UpperBound - confInt.LowerBound
+		if tightness < tightest {
+			tightest = tightness
+			tightestConfInt = confInt
+		}
+	}
+	return tightestConfInt, nil
+}
+
+// computeConfidenceIntervalSplit computes a confidence interval for mean with a given 1 - countAlpha
+// confidence level for count. It computes the corresponding alpha for sum to return a confidence interval
+// with a confidence level of at least 1 - meanAlpha. The computation is based exclusively on the confidence
+// intervals for sum and count. Thus no privacy budget is consumed by this operation.
+//
+// Result() needs to be called before ComputeConfidenceInterval, otherwise this will return an error.
+func (bm *BoundedMeanFloat64) computeConfidenceIntervalSplit(meanAlpha, countAlpha float64) (noise.ConfidenceInterval, error) {
+	if !bm.resultReturned {
+		return noise.ConfidenceInterval{}, fmt.Errorf("Result() must be called before calling computeConfidenceIntervalSplit()")
+	}
+	if err := checkSplit(meanAlpha, countAlpha); err != nil {
+		return noise.ConfidenceInterval{}, err
+	}
+
+	sumAlpha := (meanAlpha - countAlpha) / (1 - countAlpha)
+	sumConfInt, err := bm.normalizedSum.ComputeConfidenceInterval(sumAlpha)
+	if err != nil {
+		return noise.ConfidenceInterval{}, err
+	}
+	countConfInt, err := bm.count.ComputeConfidenceInterval(countAlpha)
+	if err != nil {
+		return noise.ConfidenceInterval{}, err
+	}
+
+	// BoundedMean maximizes the denominator (count) to 1.
+	countConfInt.LowerBound = math.Max(countConfInt.LowerBound, 1)
+	countConfInt.UpperBound = math.Max(countConfInt.UpperBound, 1)
+
+	// Estimate the lower and upper bounds as minimum and maximum values for mean using lower and
+	// upper bounds of confidence intervals for sum and count.
+	var meanLowerBound, meanUpperBound float64
+	if sumConfInt.LowerBound >= 0 {
+		meanLowerBound = sumConfInt.LowerBound / countConfInt.UpperBound
+	} else {
+		meanLowerBound = sumConfInt.LowerBound / countConfInt.LowerBound
+	}
+	if sumConfInt.UpperBound >= 0 {
+		meanUpperBound = sumConfInt.UpperBound / countConfInt.LowerBound
+	} else {
+		meanUpperBound = sumConfInt.UpperBound / countConfInt.UpperBound
+	}
+	// We are using normalizedSum so we need to add midPoint to both bounds.
+	meanLowerBound, meanUpperBound = meanLowerBound+bm.midPoint, meanUpperBound+bm.midPoint
+
+	// Clamp mean bounds to lower and upper bounds.
+	meanLowerBound, err = ClampFloat64(meanLowerBound, bm.lower, bm.upper)
+	if err != nil {
+		return noise.ConfidenceInterval{}, fmt.Errorf("Couldn't clamp lower bound for mean: %v", err)
+	}
+	meanUpperBound, err = ClampFloat64(meanUpperBound, bm.lower, bm.upper)
+	if err != nil {
+		return noise.ConfidenceInterval{}, fmt.Errorf("Couldn't clamp upper bound for mean: %v", err)
+	}
+	return noise.ConfidenceInterval{LowerBound: meanLowerBound, UpperBound: meanUpperBound}, nil
+}
+
+func checkSplit(meanAlpha, countAlpha float64) error {
```

Comment:
Hmm, i thought adding it would be a good assurance, but i think it's right that it might be unnecessary. Should i remove it altogether, @MashaTelyatnikova WDYT?

Reply:
+1 for removing it